File: app/src/main/java/com/privat/contacts/data/cache/mapper/MapperAddressDomainDb.java
package com.privat.contacts.data.cache.mapper;

import androidx.annotation.NonNull;

import com.privat.contacts.data.cache.model.AddressDb;
import com.privat.contacts.domain.model.AddressDomain;

import javax.inject.Inject;

public class MapperAddressDomainDb implements AddressDomain.Mapper<AddressDb> {
    @Inject
    MapperAddressDomainDb() {

    }

    @NonNull
    @Override
    public AddressDb map(int userId,
                         @NonNull String city,
                         @NonNull String streetName,
                         @NonNull String streetAddress,
                         @NonNull String zipCode,
                         @NonNull String state,
                         @NonNull String country,
                         double coordinateLat,
                         double coordinateLng) {
        return new AddressDb(
                userId,
                city,
                streetName,
                streetAddress,
                zipCode,
                state,
                country,
                coordinateLat,
                coordinateLng
        );
    }
}

File: app/src/main/java/com/privat/contacts/data/cache/mapper/MapperAddressDomainDb.kt
package com.privat.contacts.data.cache.mapper

import com.privat.contacts.data.cache.model.AddressDb
import com.privat.contacts.domain.model.AddressDomain
import javax.inject.Inject

class MapperAddressDomainDb @Inject internal constructor() : AddressDomain.Mapper<AddressDb> {
    override fun map(userId: Int,
                     city: String,
                     streetName: String,
                     streetAddress: String,
                     zipCode: String,
                     state: String,
                     country: String,
                     coordinateLat: Double,
                     coordinateLng: Double): AddressDb {
        return AddressDb(
                userId,
                city,
                streetName,
                streetAddress,
                zipCode,
                state,
                country,
                coordinateLat,
                coordinateLng
        )
    }
}
